java/vector/src/main/java/org/apache/arrow/vector/ValueVector.java

```diff
@@ -272,4 +273,11 @@
    * @param from      source vector
    */
   void copyFromSafe(int fromIndex, int thisIndex, ValueVector from);
+
+  /**
+   * Compare range values in this vector and vector in visitor.
+   * @param visitor visitor which holds the vector to compare.
+   * @return true if equals, otherwise false.
+   */
+  boolean accept(RangeEqualsVisitor visitor);
```

Comment:
I think this is likely to be useful in other cases too and, shouldn't be restricted to the RangeEquals. can we make the visitor more generic using a template interface ?

public interface VectorVisitor<OUT, IN, EX extends Exception> {..}

public <OUT, IN, EX extends Exception> OUT accept(VectorVisitor<OUT, IN, EX> visitor, IN value) throws EX {
    return visitor.visit(this, value);
  }

I'm fine if you think this can be addressed in a different PR.

Reply:
Thanks for your suggestion, I would like to think about this and address in a separate issue. Since this PR pending a long time and looks a little messy with too many comments, if you have no other comments, could we get this merged? thanks!

Reply:
Let me understand
>public interface VectorVisitor<OUT, IN, EX extends Exception> {..}

This is the template interface, RangeEqualsVisitor implemented it(with IN extends ValueVector, OUT=Boolean)

>public <OUT, IN, EX extends Exception> OUT accept(VectorVisitor<OUT, IN, EX> visitor, IN value) throws EX {
return visitor.visit(this, value);
}
Has some question here, where is the accept API located? In Visitor interface or ValueVector?

Reply:
The accept API should be in ValueVector (similar to your current impl).